pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -291,37 +611,94 @@ def controlled_resource_decomp(
 
     @classmethod
     def pow_resource_decomp(cls, z) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources for an operator raised to a power.
+
+        Args:
+            z (int): the power that the operator is being raised to
+
+        Resources:
+            This operation is self-inverse, thus when raised to even integer powers acts like
+            the identity operator and raised to odd powers it produces itself.
+
+        Returns:
+            Dict[CompressedResourceOp, int]: The keys are the operators and the associated
+                values are the counts.
+        """
         return {} if z % 2 == 0 else {cls.resource_rep(): 1}
 
 
 class ResourceCNOT(qml.CNOT, re.ResourceOperator):
     r"""Resource class for the CNOT gate.
 
     Resources:
-        There is no further decomposition provided for this gate.
+        The CNOT gate is treated as a terminal gate and thus it cannot be decomposed
+        further. Requesting the resources of this gate raises a :code:`ResourcesNotDefined` error.
+
+    .. seealso:: :class:`~.CNOT`
 
     """
 
     @staticmethod
     def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Resources:
+            The CNOT gate is treated as a terminal gate and thus it cannot be decomposed
+            further. Requesting the resources of this gate raises a :code:`ResourcesNotDefined` error.
```

Comment:
Probably should just say something to the effect of, this raises a ResourceNotDefined error because it's not further decomposable.

Reply:
The second sentence gets this point across, leaving it as is.